### ISSUE TITLE
FURB142 false positive when the target variable is used beyond the `add` or `discard` call

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#18575]
### Summary

The fix for [`for-loop-set-mutations` (FURB142)](https://docs.astral.sh/ruff/rules/for-loop-set-mutations/) changes behavior when the target variable is used after the loop.

```console
$ cat >furb142.py <<'# EOF'
s = {1, 2}
for x in (2, 3):
    s.add(x)
print(x)
# EOF

$ python furb142.py
3

$ ruff --isolated check furb142.py --select FURB142 --preview --fix
Found 1 error (1 fixed, 0 remaining).

$ cat furb142.py
s = {1, 2}
s.update((2, 3))
print(x)

$ python furb142.py 2>&1 | tail -n 1
NameError: name 'x' is not defined
```

### Version

ruff 0.11.13 (5faf72a4d 2025-06-05)
